genisys/modules/firstboot/hello.py

```diff
@@ -21,29 +20,35 @@ def __init__(self: Self, config: YAMLParser):
 
     def generate(self: Self):
         content = []
-
         json_body = {
             'message' : 'hello', 
             'ip' : '$(hostname -I)',
             'hostname' : '$(hostname)'
         }
 
-        # Not sure of json.dumps() is necessary, will need testing
+        #Convert to actual JSON string
         json_object = json.dumps(json_body)
 
         content.append("#!/bin/bash")
 
+        # Building target IP for curl request
+        prefix = "https://" if "ssl" in self.config["network"] else ""
```

Comment:
In my server config right now, SSL is a subsection under the server section. So in this case you'd want to be checking for `"ssl" in self.config["network"]["server"]`. Be careful though, because the user may not have specified the server section (or specified it with no keys). If the section wasn't specified, then `self.config["network"]["server"]` will raise an exception. If it was specified, but empty, that's gonna come up as `None` which the `str in None` also raises an exception. What I found to work best so far is to use the following pattern: `self.config["network"].get("server", {}) or {}`. Basically, `.get()` returns the value of that key in the dictionary or the default value if that key isn't specified. Then, if the key is there but empty, the `.get()` call will still return `None` which is then taken care of as `None or {}`. Python evaluates `None` to false and replaces the whole expression with the value after the `or`. Effectively, that expression makes sure whether the key is either absent or empty it returns an empty dict, which `in` is defined on. Finally, I split that off into a separate assignment since `in` has a higher precedence than `or`. Putting it all together, in the files I'm working on I get something along these lines

```python
server_config = self.config["network"].get("server", {}) or {}
if "ssl" in server_config:
```

Reply:
Fix implemented 👍